Mathlib/LinearAlgebra/Basis.lean

```diff
@@ -316,6 +316,38 @@ theorem coe_map : (b.map f : ι → M') = f ∘ b :=
 
 end Map
 
+section SMul
+variable {G G'}
+variable [Group G] [Group G']
+variable [DistribMulAction G M] [DistribMulAction G' M]
+variable [SMulCommClass G R M] [SMulCommClass G' R M]
```

Comment:
Just for my own understanding, what's the rule/is there a rule preferring `[SMulCommClass G R M]` over `[SMulCommClass R G M]`?

Reply:
It's arbitrary, the rule is "pick the one that Lean complained was missing".

Reply:
😆 okay but you are (almost) always able to replace `smul_comm` by `← smul_comm` in which case it asks for the other one

Reply:
With the extra rule "avoid picking both, and use `.symm` internally to construct the other one".

Reply:
I guess aesthetically, I'd prefer to see `SMulCommClass G R M` when attaching a `G` action to something that has `R` and `M` involved in its type.

Reply:
Yeah, that's my reasoning as well in #14863